dvc/compare.py

```diff
@@ -125,9 +138,15 @@ def shape(self) -> Tuple[int, int]:
         return len(self.columns), len(self)
 
     def drop(self, *col_names: str) -> None:
-        for col_name in col_names:
-            self._keys.remove(col_name)
-            self._columns.pop(col_name)
+        to_remove = set()
+        for key in self._keys:
+            if self.is_protected(key):
+                continue
+            if any(match(d, key) for d in col_names):
+                to_remove.add(key)
+        for col in to_remove:
+            self._keys.remove(col)
+            self._columns.pop(col)
```

Comment:
Does pandas or any dataframe library support this regex filtering?
Would it be possible to do `regex.match` from outside? something as follows:

```python
cols = match_ordered(keep, td.keys())
td.project(*cols)
```

I prefer not to complicate API too much here and keep API look like that of dataframe library. All the business logics that don't make sense in a library should be better handled outside.

Reply:
Makes sense. Will move `re.match` logic to `exp show`